hashintel/hash
# H-67: Add ability to view archived pages

## 🌟 What is the purpose of this PR?




This PR adds a "View all pages" page at `/pages`, which the list of page entities a user can see. It adds a page at `/entities` which either displays all entities a user can see, or all entities of an entity type version or entity type if the optional `entityTypeIdOrBaseUrl` query param is set.

Note that `/pages` is essentially a mask of `/entities?entityTypeIdOrBaseUrl=http://localhost:3000/@system-user/types/entity-type/page/v/1`.

## 🔍 What does this change?




- makes the entities table re-usable between the `/entities` page and the entities tab on the entity type page via a `EntitiesTable` component

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



- Currently the list of entities a user can see is all the entities in the instance of HASH, as authorization has not yet been implemented.
- Currently the "Last Edited" column for page entities represents when the page entity was last edited, which does not account for edits to nested text/block entities in the page  ([internal ticket](https://linear.app/hash/issue/H-498))
- The `/entities` page is not discoverable via the UI, as we aren't linking to it from anywhere

## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Checkout the branch and login
2. Go to the `/pages` route, or click on "View all pages" in the sidebar. Check the table lists page entities as expected, and use the filters to discover archived and external pages
3. Go to the `/entities` route (I don't think we are currently linking to this anywhere). Check the table lists all entities, and use the filters to discover archived/external entities. You can also try out the `entityTypeIdOrBaseUrl` query param to transform the page into a view of that type.
4. Go to to an entity type page, and click on the "Entities" tab. Check the table still works as previously, with the addition of filters, updated styling, and the "Last Edited"/"Last Edited By" columns.

## 📹 Demo



https://github.com/hashintel/hash/assets/42802102/68f40ee1-24fd-48fd-be07-d87a270a5b5a
